pkg/pillar/cmd/downloader/downloader.go

```diff
@@ -1546,7 +1581,11 @@ func rktFetchContainerImage(ctx *downloaderContext, key string,
 			localConfigDir = ""
 			log.Infof("rktFetchContainerImage: no Auth File")
 		}
-		imageID, err = rktFetch(dsCtx.DownloadURL, localConfigDir)
+		if len(downloadURL) == 0 {
```

Comment:
This doesn't make sense.. This will fail.. Remove it.

If download fails -- We should return with proper error set in Download status
If Creation of Auth file fails - We should return with proper error set in Download status
If rktFetch fails - We should return with proper error set in Download status